pkgs/os-specific/linux/iputils/default.nix

```diff
@@ -21,62 +19,33 @@ in stdenv.mkDerivation {
     owner = "iputils";
     repo = "iputils";
     rev = "s${time}";
-    sha256 = "19rpl48pjgmyqlm4h7sml5gy7yg4cxciadxcs24q1zj40c05jls0";
+    sha256 = "0b755gv3370c0rrphx14mrsqjb396zqnsm9lsws842a4k4zrqmvi";
   };
 
-  patches = [
-    (fetchpatch {
-      name = "dont-hardcode-the-location-of-xsltproc.patch";
-      url = "https://github.com/iputils/iputils/commit/d0ff83e87ea9064d9215a18e93076b85f0f9e828.patch";
-      sha256 = "05wrwf0bfmax69bsgzh3b40n7rvyzw097j8z5ix0xsg0kciygjvx";
-    })
-    (fetchpatch {
-      name = "add-missing-idn-declarations.patch";
-      url = "https://github.com/iputils/iputils/commit/5007d7067918fb3d950d34c01d059e5222db679a.patch";
-      sha256 = "0dhgxdhjcbb2q6snm3mjp38l066knykmrx4k8rn167cizn7akpdx";
-    })
-    (fetchpatch {
-      name = "fix-ping-idn.patch";
-      url = "https://github.com/iputils/iputils/commit/25899e849aa3abc1ad29ebf0b830262a859eaed5.patch";
-      sha256 = "1bqjcdjjnc2j6indcli7s7gbbhkcaligvh94asixfrmjzkbn533n";
-    })
+  mesonFlags = [
+    "-DBUILD_TRACEROUTE6=true"
+    "-DBUILD_RARPD=true"
   ];
 
+  patches = [ ./timespec.patch ];
   prePatch = ''
-    substituteInPlace doc/custom-man.xsl \
-      --replace "http://docbook.sourceforge.net/release/xsl/current/manpages/docbook.xsl" "${docbook_xsl}/xml/xsl/docbook/manpages/docbook.xsl"
-    for xmlFile in doc/*.xml; do
-      substituteInPlace $xmlFile \
-        --replace "http://www.oasis-open.org/docbook/xml/4.4/docbookx.dtd" "${docbook_xml_dtd_44}/xml/dtd/docbook/docbookx.dtd"
+    for f in doc/{custom-man.xsl,meson.build}; do
+      substituteInPlace $f  \
+        --replace "http://docbook.sourceforge.net/release/xsl/current/manpages/docbook.xsl" "${docbook5_xsl}/share/xml/docbook-xsl-ns/manpages/docbook.xsl"
     done
+  '' +
+    # I don't know what's going on but docbook5 examples don't have `xmlns:db`
+    # and I found that dropping the `:db` part fixes manpage generation,
+    # especially by removing space/indent issues that were ugly but also
+    # managed to adds spaces before the command char ('.') resulting
+    # in raw groff commands in the output :/.
+  ''
+    sed -i 's,xmlns:db,xmlns,' doc/*xml
```

Comment:
You probably need `docbook_xsl_ns`

Reply:
Well I thought so too-- `docbook5_xsl` is the same as `docbook_xsl_ns`-- but @primeos's PR uses the `docbook_xsl` and apparently that has none of the problems I was encountering. :/.

Thanks for taking a look, though!

Reply:
Yeah, it looks like they do use the default namespace, not the explicit `db` one:

https://github.com/iputils/iputils/blob/991916bf6ee6469285621e47038ee7aeab6ce9e8/doc/rarpd.xml#L1

Reply:
And this should be `xsl-ns` as well:

https://github.com/iputils/iputils/blob/991916bf6ee6469285621e47038ee7aeab6ce9e8/doc/custom-html.xsl#L5

Reply:
See https://github.com/docbook/xslt10-stylesheets/issues/123

Reply:
The only reason it works with non-namespaced `docbook_xsl` is because there is no namespace used in the document. Docbook 5 documents should use a namespace and when that is fixed, they will need `docbook_xsl_ns`.